interface.md

```diff
@@ -198,32 +200,45 @@ Not all variants need to be supported – for example, a posterior model might n
 model.
 
 
-### Density Calculation
+### Density Evaluation
 
 Since the different “versions” of how a model is to be understood as generative or conditioned are
 to be expressed in the type or dispatch they support, there should be no need for separate functions
-`logjoint`, `loglikelihood`, etc., which force these semantic distinctions on the implementor; one
-`logdensity` should suffice for all, with the distinction being made by the capabilities of the
-concrete model instance.
+`logjoint`, `loglikelihood`, etc., which force these semantic distinctions on the implementor; we
+therefore adapt the interface of
+[DensityInterface.jl](https://github.com/JuliaMath/DensityInterface.jl).  Its main function
+`logdensityof` should suffice for variants, with the distinction being made by the capabilities of
+the concrete model instance.
 
-Note that this generalizes `logpdf`, too, since the posterior density will of course in general be
-unnormalized and hence not a probability density.
+ DensityInterface.jl also requires the trait function `hasdensity`, which is set to `true` for the
+`AbstractProbabilisticProgram` type.  Additional functions
 
-The evaluation will usually work with the internal, concrete trace type, like `VarInfo` in Turing.jl:
+```
+DensityInterface.densityof(d, x) = exp(logdensityof(d, x))
+DensityInterface.logdensity(d) = Base.Fix1(logdensity, d)
```

Comment:
Is this correct?

Reply:
No it should be `logdensityof`

Reply:
Ah, right. I'm still getting used to the `of`...

Reply:
I was confused too